### PR TITLE
EDSC-2851: Display variable selection in a tree view

### PR DESCRIPTION
**Pull request by macrouch**

### CODE REVIEW

File: static/src/js/components/ProjectPanels/VariableTreePanel.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  // eslint-disable-next-line prefer-const
+  let items = treeView === 'hierarchy' ? hierarchyMappings : keywordMappings
+
+  // If keywordMappings or hierarchyMappings don't exist, default items to the one that does exist
+  if (keywordMappings.length && !hierarchyMappings.length) items = keywordMappings
+  if (!keywordMappings.length && hierarchyMappings.length) items = hierarchyMappings
+
+  return (
+    <>
+      <ProjectPanelSection heading="Variable Selection" />
```
Comment (macrouch) [inline static/src/js/components/ProjectPanels/VariableTreePanel.js:69]: move heading prop to line below

File: static/src/js/util/tree/TreeNode.js [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+    this.level = this.props.level || 0
+    this.parentChecked = this.props.parentChecked // Used for cascading from the top down on initial render
+    this.separator = '/'
+    this.onUpdateFinished = this.props.onUpdateFinished
+
+    // Properties with default values
+    this.allItems = {}
+    this.checked = false
+    this.children = []
+    this.disabled = false
```
Comment (macrouch) [inline static/src/js/util/tree/TreeNode.js:19]: remove

File: static/src/js/util/tree/TreeNode.js [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+    // this.expanded = this.level === 1
+    this.expanded = false
+
+    // Properties derived from element attributes, default to undefined for caching
+    this.relevantAttribute = undefined
```
Comment (macrouch) [inline static/src/js/util/tree/TreeNode.js:24]: remove

File: serverless/src/getAccessMethods/computeHierarchyMappings.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+      // Search for a object with a label matching the current name
+      const foundIndex = objectToSearch.findIndex(mapping => (
+        mapping.label === name
+      ))
+
+      // If a name was not found
+      if (foundIndex === -1) {
+        workingObject.label = name
+        if (nameIndex === nameParts.length - 2) {
```
Comment (macrouch) [inline serverless/src/getAccessMethods/computeHierarchyMappings.js:43]: more comments

File: serverless/src/getAccessMethods/computeHierarchyMappings.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            mapping.label === nameParts[nameIndex + 1]
+          ))
+
+          if (nextFound === -1) {
+            // If the next name isn't found, add a new empty object to the current workingObject.children
+            // and set that new object to the workingObject
+            workingObject.children.push({})
+            workingObject = workingObject.children[workingObject.children.length - 1]
+          } else {
+            // If the next name is found, set the working object to the foundIndex of the current workingObject.children
```
Comment (macrouch) [inline serverless/src/getAccessMethods/computeHierarchyMappings.js:77]: If the next name is found, set the working object to the _nextFound_ of the current workingObject.children

File: serverless/src/getAccessMethods/__tests__/computeHierarchyMappings.test.js [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+      },
+      {
+        conceptId: 'V1237329747-EEDTEST',
+        longName: 'Altitude',
+        name: '/gt1r/geolocation/altitude_sc'
+      },
+      {
+        conceptId: 'V1237332705-EEDTEST',
+        longName: 'Photon Index Begin',
+        name: 'gt1r/geolocation/ph_index_beg' // leading slash missing on purpose
```
Comment (abbottry) [inline serverless/src/getAccessMethods/__tests__/computeHierarchyMappings.test.js:31]: Your comment doesn't match the test after updating the code.

**Review by abbottry — APPROVED** [review]

**Review by trevorlang — APPROVED** [review]

**Pull request merged**